portal-impl/src/com/liferay/portal/service/impl/GroupLocalServiceImpl.java

```diff
@@ -861,18 +861,16 @@ public Group deleteGroup(Group group) throws PortalException {
 					group.getGroupId(), RoleConstants.TYPE_SITE);
 			}
 			else {
-				if (!group.isStagingGroup() || group.isStagedRemotely()) {
```

Comment:
Hi guys,

this should be fine. There should be no assignments anyways.

But it's possible we have a crappy code somewhere that creates site role assignments in staged groups.